frontend/src/index.tsx

```diff
@@ -33,6 +35,13 @@ createRoot(document.querySelector('#root') as HTMLElement).render(
                                 path: AppRoute.SIGN_UP,
                                 element: <Auth />,
                             },
+                            {
+                                path: '',
+                                element: <ProtectedRoute />,
```

Comment:
It'd be great to move route setups from index.tsx to the Router Provider for better responsibility organization. Plus, having a way to mark routes as private, like extending Route from react-router with a `private: boolean`, would make it easier to manage. Then, we could map routes based on this property to handle public and private routes more straightforwardly.